command/agent/job_endpoint.go

```diff
@@ -196,6 +196,11 @@ func (s *HTTPServer) ValidateJobRequest(resp http.ResponseWriter, req *http.Requ
 
 	job := ApiJobToStructJob(validateRequest.Job)
 
+	// If the job priority is not set, we fallback on the defaults specified in the server config
+	if job.Priority == 0 {
+		job.Priority = s.agent.Server().GetConfig().JobDefaultPriority
+	}
```

Comment:
I see where you're going here, but we've already got a `jobCanonicalizer` in that same `nomad/job_endpoint_hooks.go` where you've updated the `jobValidator`. It calls `(*structs.Job).Canonicalize`, to which we could pass the config value.

Reply:
:exploding_head: I completely forgot about the existence of this hook! It was much simpler than I thought, and it feels way nicer this way!